src/trapezoidal.jl

```diff
@@ -0,0 +1,55 @@
+function __solvebp_call(prob::SampledIntegralProblem, alg::TrapezoidalRule; kwargs...)
+    dim = dimension(prob.dim)
+    err = Inf64
+    data = prob.y
+    grid = prob.x
+    # inlining is required in order to not allocate
+    integrand = @inline function (i)
+        # integrate along dimension `dim`, returning a n-1 dimensional array, or scalar if n=1
+        _selectdim(data, dim, i)
```

Comment:
You're right on both accounts. The day before yesterday I refactored my code to use `eachslice,`  also adding an `evalrule(y, x, weights)` method. This does no longer require `dim` to be a value type. See my recent PR to SciMLBase, removing that constraint to the inner constructor of `SampledIntegralProblem`. I haven't pushed the code yet because I want to do some more benchmarking and documenting, for which I didnt have time yet.